apps/internal/oauth/ops/accesstokens/tokens.go

```diff
@@ -183,6 +184,32 @@ type TokenResponse struct {
 	scopesComputed bool
 }
 
+func (t *TokenResponse) UnmarshalJSON(data []byte) error {
+	type Alias TokenResponse
+	aux := &struct {
+		*Alias
+	}{
+		Alias: (*Alias)(t),
+	}
+
+	if err := json.Unmarshal(data, &aux); err != nil {
+		return err
+	}
+
+	// If refresh_in is not set, compute it as (now - expires_on) / 2 if expires_on > 2 hours from now
```

Comment:
This is wrong. Imagine: 

1. I get a token with expires_on = 4h from AAD. It's 1 PM.
2. I get later need that token, I get it from cache. It's 2 PM (so 3h expiration remaining). I now get refresh_in = 1.5h
3. I need the token later too. It's 4 PM (so 1h expiration remaining). I now get refresh_in = nil. 

What should actually happen is: 

- at step 1, save the computed refresh_in value in the cache. 
- at step 2 and step 3, evaluate if a new token from AAD is needed.